### PR TITLE
Issue98 create project vault

### PR DESCRIPTION
**Pull request by gano3s**
Add vault functionality.

## Short description:
This enhancement provides a capability to safely store the project's related secret. Like discovered or obtained passwords, tokens or even generic note related to project.

## About implementation
I am using the openssl_encrypt/openssl_decrypt methods to encrypt the secret and store it in the database. The used algorithm is 'AES-256-CTR'.
I intentionaly allow to delete the secret without knowledge of password. It is because even during the development, I several times forgot the temporary password, which was used to the encryption.

Each item in vault contains flag if it should be reportable. If it is so, then the name and note about the secret are reported in the final report. I decided to not put the secret value itself into the report. It would cause to many issues: when to ask for the password? How to prevent anybody to access generated report with passwords? and similar. This way, we can report which accounts were compromised and if needed, the passwords can be extracted manually.

I tried to cover all controllers with the unit tests.

## Examples

https://user-images.githubusercontent.com/2551605/156604531-3114942c-bff0-4343-8dbc-4730fff05cef.mov

https://user-images.githubusercontent.com/2551605/156604587-a21c7d8c-ac0d-411e-ba57-ce3befe46a9a.mov

https://user-images.githubusercontent.com/2551605/156604640-1df35c3f-3345-47fa-9a23-b1b40786f96e.mov

## Related PRs
https://github.com/reconmap/web-client/pull/137
https://github.com/reconmap/documentation/pull/12

### CODE REVIEW

File: src/Controllers/Reports/CreateReportController.php [hunk 1]
```diff
@@ -81,13 +81,14 @@ public function __invoke(ServerRequestInterface $request): array
             }
```
Comment (santiagolizardo) [inline src/Controllers/Reports/CreateReportController.php:81]: This file is very messy. You have all different data sections in a long single file. I will refactor it into ReportDataComponents so we will end up with a list of data components each one with its own logic about how to gather the data and how to assign it to the template.

File: src/Repositories/MysqlRepository.php [hunk 2]
```diff
@@ -11,6 +12,12 @@
 
 abstract class MysqlRepository
 {
+    protected ?Logger $logger = null;
+    public function setLogger(Logger $logger): void
+    {
+        $this->logger = $logger;
+    }
+
```
Comment (santiagolizardo) [inline src/Repositories/MysqlRepository.php:20]: Wondering if we should have a `logs/database.log` to log all queries when the application is in debug mode or similar.

**Review by santiagolizardo — APPROVED** [review]

File: src/Controllers/Reports/CreateReportController.php (hunk 1 above)
Comment (gano3s) [reply src/Controllers/Reports/CreateReportController.php:81]: Yeah, I agree. I also wanted to discuss how to approach this. I was thinking about preparing more creating a slightly different logic when the project is OWASP.  So, composing from components and putting it together based on the project's type makes sense to me.

File: src/Repositories/MysqlRepository.php (hunk 2 above)
Comment (gano3s) [reply src/Repositories/MysqlRepository.php:20]: This is a great idea. It would help me to identify few bugs sooner :D

**Pull request merged**